lib/miq_expression.rb

```diff
@@ -740,46 +740,31 @@ def self.get_cols_from_expression(exp, options = {})
 
   def self.get_col_info(field, options = {})
     result ||= {:data_type => nil, :virtual_reflection => false, :virtual_column => false, :sql_support => true, :excluded_by_preprocess_options => false, :tag => false, :include => {}}
-    col = field.split("-").last if field.include?("-")
-    parts = field.split("-").first.split(".")
-    model = parts.shift
 
-    if model.downcase == "managed" || parts.last == "managed"
-      result[:data_type] = :string
+    f = parse_field_or_tag(field)
```

Comment:
The purpose of this PR:

Have the `Field` object returned by `parse_field_or_tag` to have all the same information that is currently returned by `get_col_info`.